client/rpc-servers/src/lib.rs

```diff
@@ -86,31 +89,28 @@ pub async fn start_http<M: Send + Sync + 'static>(
 	metrics: Option<RpcMetrics>,
 	rpc_api: RpcModule<M>,
 	rt: tokio::runtime::Handle,
-) -> Result<HttpServerHandle, Box<dyn StdError + Send + Sync>> {
-	let max_payload_in = payload_size_or_default(max_payload_in_mb);
-	let max_payload_out = payload_size_or_default(max_payload_out_mb);
-
-	let mut acl = AccessControlBuilder::new();
-
-	if let Some(cors) = cors {
-		// Whitelist listening address.
-		// NOTE: set_allowed_hosts will whitelist both ports but only one will used.
-		acl = acl.set_allowed_hosts(format_allowed_hosts(&addrs[..]))?;
-		acl = acl.set_allowed_origins(cors)?;
-	};
-
-	let builder = HttpServerBuilder::new()
-		.max_request_body_size(max_payload_in as u32)
-		.max_response_body_size(max_payload_out as u32)
-		.set_access_control(acl.build())
-		.health_api("/health", "system_health")?
-		.custom_tokio_runtime(rt);
+) -> Result<ServerHandle, Box<dyn StdError + Send + Sync>> {
+	let max_payload_in = payload_size_or_default(max_payload_in_mb) as u32;
+	let max_payload_out = payload_size_or_default(max_payload_out_mb) as u32;
+	let host_filter = hosts_filter(cors.is_some(), &addrs);
+	let cors = try_into_cors(cors)?;
+
+	let middleware = tower::ServiceBuilder::new()
+		// Proxy `GET /health` requests to internal `system_health` method.
+		.layer(ProxyGetRequestLayer::new("/health", "system_health")?)
+		.layer(cors.clone());
+
+	let builder = ServerBuilder::new()
+		.max_request_body_size(max_payload_in)
+		.max_response_body_size(max_payload_out)
+		.set_host_filtering(host_filter)
+		.set_middleware(middleware)
+		.custom_tokio_runtime(rt)
+		.http_only();
```

Comment:
this runs only on HTTP and will be deprecated/removed in few releases.

/cc @PierreBesson @jam206  would be great if you can migrate to `health checks` to the `ws-port` when this is deployed that should work

I think the most reasonable is to keep `--rpc-port` and similar flags and just deprecate the `ws-related` flags.
Then make the `9944` the default port.